files/cache_instance_list.rb

```diff
@@ -33,12 +40,14 @@ def write_instance_cache(instance_list)
 
   def ec2
     @ec2 ||= begin
-      Fog::Compute.new({
-        :provider => 'AWS',
-        :aws_access_key_id      => config[:aws_access_key],
-        :aws_secret_access_key  => config[:aws_secret_access_key],
-        :region                 => config[:aws_region]
-      })
+      opts = { :provider => 'AWS', :region => config[:aws_region] }
+      if config[:credential_file]
+        Fog.credentials_path = config[:credential_file]
```

Comment:
I believe that we do need this and can't just change required from false to true.

I added -f cmd line arg because I personally prefer explicit reference to where creds are coming from, instead of relying on implicit knowledge that fog will look at some file in a home dir or will use FOG_RC env var.

That "if" is needed imho because if creds are not passed on command line, config[:aws_access_key_id] and config[:aws_secret_access_key] will both be nil and will be passed directly to Fog::Compute.new thus overriding global fog creds - so you'd be connecting to EC2 with both access key and secret set to nil.